### PR TITLE
feature/add-sign-up-login-register-layout

### PR DESCRIPTION
**Pull request by hoanghai126**
Anh review cho em voi a

https://github.com/PhucNguyenKhac/base-android-source-mock/assets/91648299/6848168b-f1ff-4535-b600-62669e2656f1

### CODE REVIEW

File: app/src/main/res/layout/fragment_sign_up.xml [hunk 1]
```diff
@@ -0,0 +1,173 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (NguyenKhacPhuc) [inline app/src/main/res/layout/fragment_sign_up.xml]: try to use constrait layout with complex UI

File: app/src/main/res/layout/layout_register.xml [hunk 2]
```diff
@@ -0,0 +1,294 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@color/colorPrimaryDark">
+
+    <include
```
Comment (NguyenKhacPhuc) [inline app/src/main/res/layout/layout_register.xml]: use toolbar of Android, and CollapseToolBar if possible

**Comment by tranconguet** [comment]
Please format code before create Merge request

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,3 +1,18 @@
-<resources>
-    <string name="app_name">Login UI</string>
-</resources>
+<resources>
+    <string name="app_name">Login UI</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (tranconguet) [inline app/src/main/res/values/strings.xml]: Should remove unused string

File: app/src/main/res/layout/fragment_sign_up.xml
Comment (tranconguet) [inline app/src/main/res/layout/fragment_sign_up.xml:1]: Please use the same convention when set id name for view. Eg. tvSignUp, rlContinuePhoneNumber, ivGoogle, etc.

File: app/src/main/res/layout/layout_register.xml [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+        android:layout_centerHorizontal="true"
+        android:orientation="horizontal"
+        android:layout_marginTop="20dp"
+        >
+
+        <TextView
+            android:id="@+id/tvRegister1"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Have An Account?"
```
Comment (tranconguet) [inline app/src/main/res/layout/layout_register.xml]: Should define string resource

File: app/src/main/res/layout/layout_toolbar.xml
Comment (tranconguet) [inline app/src/main/res/layout/layout_toolbar.xml:1]: Please use camelCase or underscore only

**Review by tranconguet — DISMISSED** [review]
Please resolve all comments

File: app/src/main/res/layout/fragment_sign_up.xml [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <ImageView
+            android:id="@+id/icLeft"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="start"
+            android:layout_marginStart="30dp"
```
Comment (thaoqs280899) [inline app/src/main/res/layout/fragment_sign_up.xml]: The values of dp should be moved to dimen file

File: app/src/main/res/layout/layout_login.xml [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+            android:text="@string/if_you_need_any_support_click_here"
+            android:textColor="@color/colorPrimary"
+            android:textSize="12sp" />
+
+        <TextView
+            android:id="@+id/tvSuggest2"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="2dp"
+            android:layout_marginTop="15dp"
```
Comment (thaoqs280899) [inline app/src/main/res/layout/layout_login.xml]: The values of margin should divisible by 4

File: app/src/main/res/layout/fragment_sign_up.xml [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@color/colorPrimaryDark"
+    tools:context=".ui.login.SignUpFragment">
+
+    <androidx.appcompat.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="80dp"
+        android:background="#000000"
```
Comment (thaoqs280899) [inline app/src/main/res/layout/fragment_sign_up.xml]: Using @color/colorPrimaryDark

File: app/src/main/res/layout/fragment_sign_up.xml [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+            android:paddingTop="15dp"
+            android:paddingBottom="15dp"
+            android:text="@string/sign_up_free"
+            android:textAlignment="center"
+            android:textAllCaps="false"
+            android:textColor="@color/colorPrimaryDark"
+            android:textSize="18sp"
+            android:textStyle="bold" />
+
+        <RelativeLayout
```
Comment (NguyenKhacPhuc) [inline app/src/main/res/layout/fragment_sign_up.xml]: use drawableLeft, no need to create new layout for this

File: app/src/main/res/drawable/blur_overlay.xml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+
+    <gradient
+        android:angle="90"
+        android:centerColor="#00ffffff"
```
Comment (NguyenKhacPhuc) [inline app/src/main/res/drawable/blur_overlay.xml]: move color to color file for future use

File: app/src/main/res/drawable/custom_image_icon_left.xml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="#000000"/>
+    <stroke android:color="#B3B2B2"
+        android:width="0.01dp"/>
```
Comment (NguyenKhacPhuc) [inline app/src/main/res/drawable/custom_image_icon_left.xml]: why use this dp? is it too small?

File: app/src/main/res/layout/fragment_sign_up.xml [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+            android:src="@drawable/icon_left"
+            tools:ignore="ContentDescription" />
+
+        <ImageView
+            android:id="@+id/imgLogo"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:layout_marginStart="8dp"
+            android:layout_marginEnd="8dp"
```
Comment (NguyenKhacPhuc) [inline app/src/main/res/layout/fragment_sign_up.xml]: should move to dimens file for future use

**Review by NguyenKhacPhuc — APPROVED** [review]

**Pull request merged**